node/src/components/storage.rs

```diff
@@ -2321,8 +2326,33 @@ impl Storage {
                 None => return Ok(FetchResponse::NotFound(sync_leap_identifier)),
             };
 
+        let highest_complete_block_header =
+            match self.get_header_with_metadata_of_highest_complete_block(&mut txn)? {
+                Some(highest_complete_block_header) => highest_complete_block_header,
+                None => return Ok(FetchResponse::NotFound(sync_leap_identifier)),
+            };
+
         // highest block and signatures are not requested
         if sync_leap_identifier.trusted_ancestor_only() {
+            if trusted_block_header.era_id().is_genesis() {
+                if let Some(signed_block_headers) = self.get_signed_block_headers(
+                    &mut txn,
+                    &trusted_block_header,
+                    &highest_complete_block_header,
+                    true,
+                )? {
+                    debug_assert!(signed_block_headers.len() <= 1);
+                    return Ok(FetchResponse::Fetched(SyncLeap {
+                        trusted_ancestor_only: true,
+                        trusted_block_header,
+                        trusted_ancestor_headers,
+                        signed_block_headers,
+                    }));
+                } else {
+                    warn!("unable to provide validator weights for era 1 to the requesting node")
+                }
+            }
+
```

Comment:
nit: just to avoid duplicating the return statement we can do:
```suggestion
            if trusted_block_header.era_id().is_genesis() {
                let signed_block_headers = if let Some(signed_block_headers) = self.get_signed_block_headers(
                    &mut txn,
                    &trusted_block_header,
                    &highest_complete_block_header,
                    true,
                )? {
                    debug_assert!(signed_block_headers.len() <= 1);
                    signed_block_headers
                } else {
                    warn!("unable to provide validator weights for era 1 to the requesting node");
                    vec![]
                }
            }
```